agent/src/main/java/com/cmclinnovations/agent/service/application/LifecycleService.java

```diff
@@ -174,7 +177,27 @@ public ResponseEntity<StandardApiResponse<?>> getContracts(String resourceID, bo
     Queue<SparqlBinding> instances = this.getService.getInstances(resourceID, null, "", additionalQueryStatement,
         requireLabel, contractVariables);
     return this.responseEntityBuilder.success(null, instances.stream()
-        .map(SparqlBinding::get)
+        .map(binding -> {
+          return (Map<String, Object>) binding.get().entrySet().stream()
+              .map(entry -> {
+                // Replace recurrence with schedule type
+                if (entry.getKey().equals(LifecycleResource.SCHEDULE_RECURRENCE_PLACEHOLDER_KEY)) {
+                  SparqlResponseField recurrence = TypeCastUtils.castToObject(entry.getValue(),
+                      SparqlResponseField.class);
+                  return new AbstractMap.SimpleEntry<>(
+                      LocalisationTranslator.getMessage(LocalisationResource.VAR_SCHEDULE_TYPE_KEY),
+                      new SparqlResponseField(recurrence.type(),
+                          LifecycleResource.getScheduleTypeFromRecurrence(recurrence.value()),
+                          recurrence.dataType(), recurrence.lang()));
+                }
+                return entry;
+              })
+              .collect(Collectors.toMap(
+                  Map.Entry::getKey,
+                  (entry -> TypeCastUtils.castToObject(entry.getValue(), Object.class)),
+                  (oldVal, newVal) -> oldVal,
+                  LinkedHashMap::new));
+        })
```

Comment:
Could be moved to the SPARQL query by adding the mapping between periods and schedule types to the ontology and then using coalesce for the default option.